app/tests/workstations_tests/test_templatetags.py

```diff
@@ -151,24 +162,55 @@ def test_workstation_query_for_images(settings):
     image, overlay = ImageFactory.build_batch(2)
     config = WorkstationConfigFactory.build()
 
-    qs = workstation_query(image=image)
+    qs = get_workstation_query_string(image=image)
     assert "&" not in qs
     assert f"{settings.WORKSTATIONS_BASE_IMAGE_QUERY_PARAM}={image.pk}" in qs
     assert (
         f"{settings.WORKSTATIONS_OVERLAY_QUERY_PARAM}={overlay.pk}" not in qs
     )
     assert f"{settings.WORKSTATIONS_CONFIG_QUERY_PARAM}={config.pk}" not in qs
 
-    qs = workstation_query(image=image, overlay=overlay)
+    qs = get_workstation_query_string(image=image, overlay=overlay)
     assert "&" in qs
     assert f"{settings.WORKSTATIONS_BASE_IMAGE_QUERY_PARAM}={image.pk}" in qs
     assert f"{settings.WORKSTATIONS_OVERLAY_QUERY_PARAM}={overlay.pk}" in qs
     assert f"{settings.WORKSTATIONS_CONFIG_QUERY_PARAM}={config.pk}" not in qs
 
-    qs = workstation_query(image=image, config=config)
+    qs = get_workstation_query_string(image=image, config=config)
     assert "&" in qs
     assert f"{settings.WORKSTATIONS_BASE_IMAGE_QUERY_PARAM}={image.pk}" in qs
     assert (
         f"{settings.WORKSTATIONS_OVERLAY_QUERY_PARAM}={overlay.pk}" not in qs
     )
     assert f"{settings.WORKSTATIONS_CONFIG_QUERY_PARAM}={config.pk}" in qs
+
+
+@pytest.mark.django_db
+def test_workstation_session_control_data():
+    wk = WorkstationFactory()
+    obj = ReaderStudyFactory()
+    with pytest.raises(TypeError):
+        workstation_session_control_data()
+        workstation_session_control_data(workstation=wk)
+        workstation_session_control_data(context_object=obj)
+
+    data = workstation_session_control_data(
+        workstation=wk,
+        context_object=obj,
+    )
+    assert "data-session-control" in data
+    url = reverse(
+        "workstations:workstation-session-create", kwargs={"slug": wk.slug}
+    )
+    assert f"data-create-session-url={url!r}" in data  # noqa B907
+    assert 'data-workstation-query=""' in data
+    assert (
+        f'data-workstation-window-identifier="workstation-{obj._meta.app_label}"'
+        in data
+    )
+
+    data2 = workstation_session_control_data(
+        workstation=wk, context_object=obj, reader_study=obj, timeout=200
+    )
+    assert f'data-workstation-query="readerStudy={obj.pk}"' in data2
+    assert 'data-timeout="200"' in data2
```

Comment:
Why this pattern? If you do `assert data == 'data-session-control data-create-session-url='http://the/url data-timeout="200"'` etc, then you catch missing spaces etc. You can drop in pks as needed.